New template added: Umami

sancho1952007 opened the pull request:
# Install Umami on Codesphere

By installing Umami on Codesphere, my objective is to establish a self-hosted and secure web analytics solution for websites. This strategic move ensures complete control over data ownership and privacy, steering away from relying on external analytics services.

Utilizing Codesphere's infrastructure allows for seamless deployment, efficient resource management, and effortless integration with Umami. The installation on Codesphere not only guarantees robust analytics for understanding user behavior and website performance but also aligns with a cost-effective and scalable approach.

Ultimately, this endeavor aims to empower valuable insights into your website's metrics while maintaining autonomy and data security.

## What I did
1. I created a ci.yml file to automate the whole installation of Umami on Codesphere.
2. Created blog posts to understand how to install it on Codesphere.

## Links
- Blog Posts:
  - Dev.to - https://dev.to/sancho1952007/host-your-own-google-analytics-alternative-using-umami-and-codesphere-2jnk
  - Medium - https://medium.com/@sanchogodinho/host-your-own-google-analytics-alternative-using-umami-and-codesphere-2353edf3a89a

- Workspace URL: https://53867-3000.2.codesphere.com/dashboard

## Checklist
- [x] My project has been thoroughly tested and is fully functional.
- [x] The repository for my project includes a `ci.yml` file (Refer to [CodeSphere CI Pipelines](https://docs.codesphere.com/getting-started/ci-pipelines)).
- [x] The blog post about my project has been published.
- [x] All workspaces for this project are deployed on Codesphere using the CI pipeline.
- [x] My app is persistently installed in a workspace (all components are installed and configured somewhere below the `/home/user/app` path).
- [x] My pull request is well-written and clear.

Check the README.md or any blog post mentioned here to get more details on the installation of Umami on Codesphere.

Datata1 commented on Umami/README.md:
```diff
@@ -0,0 +1,42 @@
+## Steps for installing Umami on Codesphere.
+1. Signup On Codesphere — https://codesphere.com/
+2. Create a new instance using the “+ New Workspace” button.
+3. Put in the URL: https://github.com/sancho1952007/install-umami-on-codesphere.git in the input box.
```
Instead of a reference to a repository of yours, change it to https://github.com/codesphere-community/umami

sancho1952007 replied:
Done ✅

Datata1 commented on Umami/README.md:
```diff
@@ -0,0 +1,42 @@
+
+You're ready to now use it on your site!
+
+Also, If the restart fails in the future, Follow step 9, then directly go to the Run section, and then click the "Run" button.
```
Does the restart fail sometimes? What is the reason why the restart fails?

sancho1952007 replied:
Just a fallback in case of Codesphere's downtime or failure to start the ci.yml's Run command.

Datata1 commented on Umami/ci.yml:
```diff
@@ -0,0 +1,23 @@
+prepare:
+  steps:
+    - name: Install n
```
n is already preinstalled in a codesphere workspace

Datata1 commented on Umami/ci.yml:
```diff
@@ -0,0 +1,23 @@
+      command: sudo n prune
+    - name: install latest NPM
+      command: sudo npm i -g npm
+    - name: Install Yarn
```
yarn is already preinstalled in a codesphere workspace

sancho1952007 replied:
Removed This ✅

Datata1 commented on Umami/ci.yml:
```diff
@@ -0,0 +1,23 @@
+      command: sudo n stable
+    - name: Delete older nodejs version because it clashes in the Run CI command.
+      command: sudo n prune
+    - name: install latest NPM
```
updating npm is not mandatory

sancho1952007 replied:
Removed This ✅

Datata1 reviewed (changes requested):
about your blogposts: 
- Please change the reference to your personal repository to https://www.github.com/codesphere-community/umami

Datata1 commented:
Hello Sancho,

I have reviewed your code and have a few requested changes.

We discussed your blog post, and it appears to be a bit short and lacking in storytelling. It would be beneficial if you could showcase A/B testing on Codesphere and analyze the differences using Umami, so that the blog post doesn't just focus on deploying without any context.

Here are some resources that might help:

[A/B Testing with Codesphere and PostHog](https://codesphere.com/articles/how-to-a-b-testing-with-codesphere-and-posthog)
[Umami Documentation on Sending Stats](https://umami.is/docs/sending-stats)
[A/B Testing Using PostHog](https://codesphere.com/articles/a-b-testing-using-posthog)

You can refer to the blog posts above to set up an A/B test on Codesphere. If you need assistance, please open a new support thread in our Discord Community channel to streamline the support process.

Additionally, I encourage you to include pictures to enhance the reader's experience. Here are some examples of how we want these blog posts to look:
- [Visual Studio Code Sync](https://codesphere.com/articles/visual-studio-code-sync)
- [n8n: A Cost-Saving Zapier Alternative Hosted on Codesphere](https://codesphere.com/articles/n8n-a-cost-saving-zapier-alternative-hosted-on-codesphere-2)
- [Build a To-Do Application with Angular](https://codesphere.com/articles/build-a-to-do-application-with-angular)
Let me know if you need further assistance or clarification.

sancho1952007 commented:
Hey, I've made the following changes.
1. Added some more information about Umami and some comparisons to popular alternatives in my blogs.
2. Updated the ci.yml file to resolve the issues you pointed out.
3. Created a video demonstrating how to install it on your workspace.

sancho1952007 commented:
All Required Changes Are Made!

Datata1 commented:
Hey Sancho,

do you know how to set up A/B tests on Codesphere? It would increase the quality of your blogpost if you would showcase a specific usecase for umami. 

- https://codesphere.com/articles/how-to-a-b-testing-with-codesphere-and-posthog
- https://codesphere.com/articles/a-b-testing-using-posthog
- https://codesphere.com/articles/all-the-math-you-need-to-conduct-an-a-b-test
- https://docs.codesphere.com/getting-started/a-b-testing

Datata1 commented:
Hey Sancho,

Thank you for your contribution. I have some suggestions for improving your blog post:

- Point 2: It's not just a website you want to track, but rather different versions of a website.
- Point 2: Clarify which tracking code you're referring to, where it can be obtained, and its purpose.
- Point 4: Instead of mentioning two separate websites, emphasize that these are distinct versions of the same website.
- Point 4: Provide a brief explanation of why additional tracking data is necessary for an A/B test. Explain why the approach differs from the automatic tracking described in Umami's documentation ([tracker functions](https://umami.is/docs/tracker-functions)). 
- Point 4: Question the necessity of opening a new head tag in your workspace project for this purpose.

I recommend rewriting these instructions for A/B testing, with a focus on points 2-4. Additionally, consider adding example code for the tracking implementation, similar to the approach used in this blog post: [Building a Countdown Webpage for the Codesphere Hackathon](https://codesphere.com/articles/building-a-countdown-webpage-for-the-codesphere-hackathon).

Keep in mind that your target audience may be unfamiliar with Umami, so it's important to provide **detailed explanations**. Utilizing visual elements beyond plain text can also enhance reader engagement with the blog post.

sancho1952007 commented:
You can merge the pull request!

Simon-Count reviewed (approved):
Approving the changes. Readme might still require small changes later but we can do that later.
